
 This patch fixes a problem where the Options are not properly passed
 from the last Option group to the NULL option group -- i.e. some of
 the files do not get the correct options.

 Apply to Bacula 1.34.6 with:
 
 cd <bacula-source>
 patch -p0 <1.34.6-find.patch
 make
 ...

Index: src/findlib/find.c
===================================================================
RCS file: /cvsroot/bacula/bacula/src/findlib/find.c,v
retrieving revision 1.18
diff -u -r1.18 find.c
--- src/findlib/find.c	9 Jun 2004 11:52:24 -0000	1.18
+++ src/findlib/find.c	6 Aug 2004 13:25:52 -0000
@@ -160,10 +160,10 @@
 
    for (j=0; j<incexe->opts_list.size(); j++) {
       findFOPTS *fo = (findFOPTS *)incexe->opts_list.get(j);
+      ff->flags = fo->flags;
+      ff->GZIP_level = fo->GZIP_level;
       for (k=0; k<fo->wild.size(); k++) {
 	 if (fnmatch((char *)fo->wild.get(k), ff->fname, 0) == 0) {
-	    ff->flags = fo->flags;
-	    ff->GZIP_level = fo->GZIP_level;
 	    if (ff->flags & FO_EXCLUDE) {
 	       return false;	      /* reject file */
 	    }
@@ -205,7 +205,6 @@
    case FT_NOFOLLOW:
    case FT_NOSTAT:
    case FT_NOCHG:
-   case FT_DIRNOCHG:
    case FT_ISARCH:
    case FT_NORECURSE:
    case FT_NOFSCHG:
@@ -222,6 +221,7 @@
    case FT_RAW:
    case FT_FIFO:
    case FT_SPEC:
+   case FT_DIRNOCHG:
       if (accept_file(ff)) {
 	 return ff->callback(ff, hpkt);
       } else {
